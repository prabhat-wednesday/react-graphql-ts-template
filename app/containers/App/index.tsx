/**
 *
 * App.js
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 *
 */
import React from 'react';
import { withRouter } from 'react-router';
import { compose } from 'redux';
import map from 'lodash-es/map';
import { ThemeProvider } from 'styled-components';
import { colors } from '@app/themes';
import { HEADER_HEIGHT, MIN_SIDEBAR_WIDTH } from '@app/utils/constants';
import { For } from '@app/components';
import { routeConfig } from '@app/routeConfig';
import { Route, Switch } from 'react-router-dom';
const theme = {
  fg: colors.primary,
  bg: colors.secondaryText,
  headerHeight: HEADER_HEIGHT,
  sidebarWidth: MIN_SIDEBAR_WIDTH
};

export function App() {
  return (
    <ThemeProvider theme={theme}>
      <For
        ParentComponent={(props) => <Switch {...props} />}
        of={map(Object.keys(routeConfig))}
        renderItem={(routeKey, index) => {
          const Component = routeConfig[routeKey].component;
          return (
            <Route
              exact={routeConfig[routeKey].exact}
              key={index}
              path={routeConfig[routeKey].route}
              render={(props) => {
                const updatedProps = {
                  ...props,
                  ...routeConfig[routeKey].props
                };
                return <Component {...updatedProps} />;
              }}
            />
          );
        }}
      />
    </ThemeProvider>
  );
}

export default compose(withRouter)(App);
