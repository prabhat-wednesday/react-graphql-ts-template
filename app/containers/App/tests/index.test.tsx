import React from 'react';
import { renderProvider } from '@utils/testUtils';
import { AppTest as App } from '@containers/App';
import { BrowserRouter } from 'react-router-dom';

describe('<App /> container tests', () => {
  it('should render and match the snapshot', async () => {
    const { baseElement } = renderProvider(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
