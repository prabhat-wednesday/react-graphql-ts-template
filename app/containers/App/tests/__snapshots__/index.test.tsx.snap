// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> container tests should render and match the snapshot 1`] = `
<div>
  <div>
    <div
      class="InputSearchBox__InputContainer-d9b2by-0 jSCzWJ"
    >
      <label
        class="InputSearchBox__CustomLabel-d9b2by-1 lnilR"
      >
        Search your favourite singer
      </label>
      <input
        class="ant-input InputSearchBox__CustomInput-d9b2by-2 vkirg"
        placeholder="search here"
        type="text"
        value=""
      />
    </div>
    <div
      class="ant-row ItuneCard__CustomRow-sc-1g3ywlw-0 biCXNC"
    />
  </div>
</div>
`;
